### ISSUE TITLE
"trying to overwrite `/._.', which is also in package base-files" while trying to bootstrap 0.45.0 on macOS 10.14.6

### ISSUE THREAD

**Issue by clepple** [fink/fink#202]
Attempting to set up a clean build environment on 10.14.6 in `/opt/sw` to address fink/fink-distributions#499 (avoiding `/sw` to potentially smooth the 10.14->10.15 upgrade process), and I ran into this:

```
Unpacking fink (from .../fink_0.45.0-141_darwin-x86_64.deb) ...
/opt/sw/bootstrap/bin/dpkg: error processing /opt/sw/fink/dists/stable/main/binary-darwin-x86_64/base/fink_0.45.0-141_darwin-x86_64.deb (--install):
 trying to overwrite `/._.', which is also in package base-files
/opt/sw/bootstrap/bin/dpkg-deb: subprocess paste killed by signal (Broken pipe: 13)
Errors were encountered while processing:
 /opt/sw/fink/dists/stable/main/binary-darwin-x86_64/base/fink_0.45.0-141_darwin-x86_64.deb
### execution of /opt/sw/bootstrap/bin/dpkg failed, exit code 1
can't install package fink-0.45.0-141

ERROR: Bootstrap phase 2 failed!
```

Disk Utility says `/` is on an APFS volume (with FileVault enabled), and it appears to be case-insensitive.

Both `/._.` and `/._opt` seem to be created as part of the bootstrap process.

Should I inject from a specific branch on git, or should this work with the fink-0.45.0 tarball?

Also, if this is going to be too much of a hassle to track down, I can try to build on a slower 10.15 machine, or upgrade this one.

**Comment by nieder** [comment]
Do you have some weird tar, zip, or other utility someplace in /usr or /usr/local?  `._FOO` files are usually where programs stick resource forks or other extended properties created in HFS+ or APFS. If you do have something installed that doesn't know about these extended properties and is pre-empting the system tools, the extended properites will be shoved into a `._FOO` file, which is how things are being stored in your .deb files and causing the collision.

**Comment by clepple** [comment]
@dmacks, thanks for transferring.

@nieder, not that I know of. I removed the old Fink initialization lines from the dot files, opened a new terminal, and checked the path. I will poke around to see if there's a way to verify the binaries in /usr/bin.

The `._FOO` files in `/` reappear after removing them.

I'm not sure what the pattern is in the other directories under `/opt/sw`. I don't see any `._FOO` files in `/opt/sw/bootstrap`, but they are in other directories like `/opt/sw/etc`.

```
$ tar --version
bsdtar 2.8.3 - libarchive 2.8.3
$ zip --version
Copyright (c) 1990-2008 Info-ZIP - Type 'zip "-L"' for software license.
This is Zip 3.0 (July 5th 2008), by Info-ZIP.
...
```

**Comment by clepple** [comment]
Is there a recommended way to debug `./bootstrap`? I tried using `fs_usage`, but it's hard to scope it, and I didn't see much indicating when the `._foo` files were created (just when they were first accessed).

On a related note, should `/usr/bin/tar` be a symlink to `bsdtar`? At what point should `./bootstrap` start using the locally-compiled GNU tar?

**Comment by clepple** [comment]
Here's the really weird part: `./bootstrap /sw2` built fine, even with some leftover `/._.` and `/._opt` files. So it looks like I have a way forward for updating packages, but it seems odd that the two paths behave differently.

**Comment by nieder** [comment]
`/` is APFS. Is there any chance that /opt is on a different volume?

**Comment by nieder** [comment]
```
$ ls -l /usr/bin/tar /usr/bin/bsdtar 
-rwxr-xr-x  1 root  wheel  71024 Mar 27  2018 /usr/bin/bsdtar*
lrwxr-xr-x  1 root  wheel      6 Jan  6  2019 /usr/bin/tar@ -> bsdtar
```
So that looks normal.

The ._foo files are being created by the filesystem when it's trying to write some extended attribute that is normally linked internally (used to be resource fork, now more commonly an ACL or similar) and either the utility or the underlying filesystem where the file is located doesn't understand it and then gets dumped as a ._foo file. That's also why some ZIP files sometimes end up with a `_MACOSX` folder, where those metachunks get tossed. Apple-zip knows how to handle them, but I don't think the ZIP format itself does, hence the split.

**Comment by clepple** [comment]
Same device according to `df`:

```
$ df / /opt
Filesystem    512-blocks       Used  Available Capacity  iused               ifree %iused  Mounted on
/dev/disk2s1  2189683984 1855246168  309819064    86%  4172598 9223372036850603209    0%   /
/dev/disk2s1  2189683984 1855246168  309819064    86%  4172598 9223372036850603209    0%   /
```

**Comment by nieder** [comment]
@clepple Did this ever resolve itself? Or is /opt/sw still failing while /sw2 (or is it /opt/sw2?) works?

**Comment by clepple** [comment]
@nieder nope, did not resolve itself - I just started using /sw2, which works well enough. I still need to set aside a chunk of time to test #543. If others are still able to bootstrap into /opt/sw on 10.15+, I am fine with "closed, won't fix" since 10.14 is getting long in the tooth.

**Comment by nieder** [comment]
If /sw2 works on the same machine, the only thing I can think of is that something changed some FS bit for /opt, triggering the behavior you're seeing. My only further suggestion is to check `GetFileInfo /sw2` vs `GetFileInfo /opt` vs `GetFileInfo /opt/sw` .

**Comment by clepple** [comment]
```sh
$ GetFileInfo /sw2
directory: "/sw2"
attributes: avbstclinmedz
created: 11/13/2019 08:47:14
modified: 12/30/2019 12:23:45
$ GetFileInfo /opt
directory: "/opt"
attributes: aVbstclinmedz
created: 05/17/2014 01:57:00
modified: 11/13/2019 07:21:04
$ GetFileInfo /opt/sw
directory: "/opt/sw"
attributes: avbstclinmedz
created: 11/13/2019 07:19:54
modified: 11/13/2019 07:21:04
```

**Comment by nieder** [comment]
That all looks correct. The uppercase `V` makes `/opt` invisible in Finder, but that's normal.
